numba/core/dispatcher.py

```diff
@@ -847,11 +856,17 @@ def compile(self, sig):
         with ExitStack() as scope:
             cres = None
 
-            def cb(dur):
+            def cb_compiler(dur):
                 if cres is not None:
                     self._callback_add_compiler_timer(dur, cres)
 
-            scope.enter_context(ev.install_timer("numba:compiler_lock", cb))
+            def cb_llvm(dur):
+                if cres is not None:
+                    self._callback_add_llvm_timer(dur, cres)
```

Comment:
Should this callback generating pattern be a utility function, same code occurring I think 4 or 5 times in this module?

Reply:
The `cres` is a non-local that makes it impossible to be just a utility function. It will have a extra class just to do this.